src/LCCheating/CheatingParticleIDAlgorithm.cc

```diff
@@ -0,0 +1,105 @@
+/**
+ *  @file   PandoraSDK/src/Templates/CheatingParticleIDAlgorithm.cc
+ * 
+ *  @brief  Implementation of the cheating particle id algorithm class.
+ * 
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "LCCheating/CheatingParticleIDAlgorithm.h"
+
+using namespace pandora;
+
+CheatingParticleIDAlgorithm::CheatingParticleIDAlgorithm() : 
+    m_useClusterOverTrackID(true)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode CheatingParticleIDAlgorithm::Run()
+{
+    const PfoList *pPfoList = NULL;
+    PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::GetCurrentList(*this, pPfoList));
+
+    for (const ParticleFlowObject *const pPfo : *pPfoList)
+    {
+        try
+        {
+            const ClusterList clusterList = pPfo->GetClusterList();
+            const TrackList trackList = pPfo->GetTrackList();
+
+            const int nClusters(clusterList.size());
+            const int nTracks(trackList.size());
+
+            if (nTracks > 1) 
```

Comment:
Is this really a failure? What about particles for which the V0 finder has declared sibling (e.g. e+ and e-) tracks? Having said that, do we really want an exhaustive treatment for Pfos with multiple tracks? You might want to handle the 2 track case, but my point is that it's more a limitation of the algorithm, rather than some failure in the input Pfo.